src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -250,6 +261,29 @@ void MessageGenerator::Generate(io::Printer* printer) {
   GenerateMessageSerializationMethods(printer);
   GenerateMergingMethods(printer);
 
+  if (has_extension_ranges_) {
```

Comment:
Is it possible to: Make IExtensionMessage a abstract class, put  _extensions and the following implementation into IExtensionMessage (IExtendableMessage)?  I think we used to have a similar discussion on IMessage, not sure if they have the same situation.

Then no need to generate the duplicate code. For other generated code that need to use _extensions, add a getExtensionSet() in IExtensionMessage or more methods.

Reply:
I actually experimented with this at one point. I decided against it at the time since it was difficult to do and didn't really payoff, there were too many choices with how to implement each with their own costs. I'd prefer to keep it simple with interfaces even if it has a little code duplication.

Reply:
acknowledged